### PR TITLE
Method findBy with a field of type List cannot be implemented

### PR DESCRIPTION
**Pull request by radovanradic**
Fix for #1909

### CODE REVIEW

File: data-processor/src/main/java/io/micronaut/data/processor/visitors/finders/AbstractCriteriaMethodMatch.java [hunk 1]
```diff
@@ -556,6 +556,12 @@ private boolean isValidType(ClassElement genericType, SourcePersistentProperty p
         if (TypeUtils.areTypesCompatible(genericType, property.getType()) && !TypeUtils.isObjectClass(genericType)) {
             return true;
         }
+        if (TypeUtils.isContainerType(property.getType())) {
```
Comment (radovanradic) [inline data-processor/src/main/java/io/micronaut/data/processor/visitors/finders/AbstractCriteriaMethodMatch.java:559]: Without this fix, property tags can be array of `String` but not `List<String>` so I think it makes sense checking this too. It did not break anything else so should be safe.

File: data-document-model/src/main/java/io/micronaut/data/document/model/query/builder/MongoQueryBuilder.java [hunk 2]
```diff
@@ -218,6 +218,19 @@ public final class MongoQueryBuilder implements QueryBuilder {
         addCriterionHandler(QueryModel.Contains.class, (context, obj, criterion) -> {
             handleRegexPropertyExpression(context, obj, criterion, criterion.isIgnoreCase(), false, false, false);
         });
+        addCriterionHandler(QueryModel.ArrayContains.class, (context, obj, criterion) -> {
```
Comment (dstepanov) [inline data-document-model/src/main/java/io/micronaut/data/document/model/query/builder/MongoQueryBuilder.java:221]: `ArrayContains` might be a bit confusing for people not using an array as the actual type but a collection. Maybe we can just have `contains` and decide based on the type/association?

File: data-mongodb/src/test/resources/logback.xml [hunk 3]
```diff
@@ -9,7 +9,7 @@
         </encoder>
     </appender>
 
-    <root level="warn">
+    <root level="debug">
```
Comment (dstepanov) [inline data-mongodb/src/test/resources/logback.xml:12]: Probably unwanted

File: src/main/docs/guide/azureCosmos/azureCosmosCriteriaSpecifications/azureCosmosCriteriaExecuteQuery.adoc [hunk 4]
```diff
@@ -62,7 +62,7 @@ Then you can combine them for `find` or `count` queries:
 
 snippet::example.PersonRepositorySpec[project-base="doc-examples/azure-cosmos-example",source="test" tags="find",indent="0"]
 
-Specific criteria supported only by Micronaut Azure Cosmos Data is `ARRAY_CONTAINS` and for a class having an array of strings named `tags` it can be used either via custom repository method like this:
+Specific criteria supported Micronaut Azure Cosmos Data is `ARRAY_CONTAINS` and for a class having an array of strings named `tags` it can be used either via custom repository method like this:
```
Comment (dstepanov) [inline src/main/docs/guide/azureCosmos/azureCosmosCriteriaSpecifications/azureCosmosCriteriaExecuteQuery.adoc:65]: Please also add it to https://micronaut-projects.github.io/micronaut-data/latest/guide/#criteria

**Review by dstepanov — CHANGES_REQUESTED** [review]

File: data-mongodb/src/test/resources/logback.xml (hunk 3 above)
Comment (radovanradic) [reply data-mongodb/src/test/resources/logback.xml:12]: Yes, but I wanted also to see if there would be some failures on CI but will remove it before merging.

File: src/main/docs/guide/azureCosmos/azureCosmosCriteriaSpecifications/azureCosmosCriteriaExecuteQuery.adoc (hunk 4 above)
Comment (radovanradic) [reply src/main/docs/guide/azureCosmos/azureCosmosCriteriaSpecifications/azureCosmosCriteriaExecuteQuery.adoc:65]: Didn't find good place where to put it but will add it somewhere.

File: data-document-model/src/main/java/io/micronaut/data/document/model/query/builder/MongoQueryBuilder.java (hunk 2 above)
Comment (radovanradic) [reply data-document-model/src/main/java/io/micronaut/data/document/model/query/builder/MongoQueryBuilder.java:221]: We already have Contains 
```
addCriterionHandler(QueryModel.Contains.class, (context, obj, criterion) -> {
            handleRegexPropertyExpression(context, obj, criterion, criterion.isIgnoreCase(), false, false, false);
        });
```
ant that is used for Strings everywhere. On the other hand we have already ArrayContains for Azure Cosmos Db so it makes sense to call it the same here I think, we can explain in the docs and examples how to use it with list or array.

File: src/main/docs/guide/azureCosmos/azureCosmosCriteriaSpecifications/azureCosmosCriteriaExecuteQuery.adoc (hunk 4 above)
Comment (radovanradic) [reply src/main/docs/guide/azureCosmos/azureCosmosCriteriaSpecifications/azureCosmosCriteriaExecuteQuery.adoc:65]: Added some docs

File: data-mongodb/src/test/resources/logback.xml (hunk 3 above)
Comment (radovanradic) [reply data-mongodb/src/test/resources/logback.xml]: Removed

File: src/main/docs/guide/azureCosmos/azureCosmosCriteriaSpecifications/azureCosmosCriteriaExecuteQuery.adoc (hunk 4 above)
Comment (dstepanov) [reply src/main/docs/guide/azureCosmos/azureCosmosCriteriaSpecifications/azureCosmosCriteriaExecuteQuery.adoc:65]: It should be in that table with all possible criteria, with a note that it's only for some DBs

Comment (radovanradic) [reply src/main/docs/guide/azureCosmos/azureCosmosCriteriaSpecifications/azureCosmosCriteriaExecuteQuery.adoc]: Added

File: data-document-model/src/main/java/io/micronaut/data/document/model/query/builder/MongoQueryBuilder.java (hunk 2 above)
Comment (dstepanov) [reply data-document-model/src/main/java/io/micronaut/data/document/model/query/builder/MongoQueryBuilder.java:221]: Ok so maybe we can also have the alias `collectionContains`

Comment (radovanradic) [reply data-document-model/src/main/java/io/micronaut/data/document/model/query/builder/MongoQueryBuilder.java:221]: Added, hoping it won't cause more confusion:/

**Pull request merged**